x/market/keeper/deal.go

```diff
@@ -0,0 +1,115 @@
+package keeper
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	gogotypes "github.com/gogo/protobuf/types"
+	"github.com/medibloc/panacea-core/v2/x/market/types"
+)
+
+func (k Keeper) CreateNewDeal(ctx sdk.Context, owner sdk.AccAddress, deal types.Deal) (uint64, error) {
+	dealId := k.GetNextDealNumberAndIncrement(ctx)
+
+	newDeal, err := k.newDeal(dealId, deal)
+	if err != nil {
+		return 0, err
+	}
+
+	var coins sdk.Coins
+	coins = append(coins, *deal.GetBudget())
+
+	acc := k.accountKeeper.GetAccount(ctx, sdk.AccAddress(newDeal.GetDealAddress()))
+	if acc != nil {
+		return 0, sdkerrors.Wrapf(types.ErrDealAlreadyExist, "deal %d already exist", dealId)
+	}
+
+	k.SetDeal(ctx, newDeal)
+
+	acc = k.accountKeeper.NewAccount(ctx, authtypes.NewModuleAccount(
+		authtypes.NewBaseAccountWithAddress(
+			sdk.AccAddress(newDeal.GetDealAddress()),
+		),
+		newDeal.DealAddress),
+	)
+	k.accountKeeper.SetAccount(ctx, acc)
+
+	err = k.bankKeeper.SendCoins(ctx, owner, sdk.AccAddress(newDeal.GetDealAddress()), coins)
+	if err != nil {
+		return 0, nil
```

Comment:
```suggestion
		return 0, err
```
I think the second return will be an err.
And wrapping error message with `The owner's insufficient balance.`